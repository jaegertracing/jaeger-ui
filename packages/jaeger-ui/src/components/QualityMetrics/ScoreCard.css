/*
Copyright (c) 2020 The Jaeger Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.ScoreCard {
  margin: 10px;
  max-width: 25%;
  min-width: 180px;
  text-align: center;
}

.ScoreCard--CircularProgressbarWrapper {
  padding: 3px;
}

.ScoreCard--TitleHeader {
  border-bottom: solid 1px #ddd;
  box-shadow: 0px 5px 5px -5px rgba(0, 0, 0, 0.3);
  font-size: 1.8em;
}
