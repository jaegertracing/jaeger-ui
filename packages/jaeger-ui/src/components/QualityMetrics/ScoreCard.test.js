// Copyright (c) 2020 Uber Technologies, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import React from 'react';
import ScoreCard from './ScoreCard';
import shallow from '../../utils/ReactShallowRenderer.test.js';

describe('ScoreCard', () => {
  const link = 'test.link';
  const label = 'Test Score';
  const value = 42;
  const max = 108;

  it('renders as expected when score is below max', () => {
    expect(
      shallow(
        <ScoreCard
          link={link}
          score={{
            label,
            max,
            value,
          }}
        />
      )
    ).toMatchSnapshot();
  });

  it('renders as expected when score is max', () => {
    expect(
      shallow(
        <ScoreCard
          link={link}
          score={{
            label,
            max,
            value: max,
          }}
        />
      )
    ).toMatchSnapshot();
  });

  it('renders as expected when score is zero', () => {
    expect(
      shallow(
        <ScoreCard
          link={link}
          score={{
            label,
            max,
            value: 0,
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
