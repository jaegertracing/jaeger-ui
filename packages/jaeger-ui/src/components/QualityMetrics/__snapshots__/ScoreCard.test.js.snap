// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCard renders as expected when score is below max 1`] = `
<div
  className="ScoreCard"
>
  <span
    className="ScoreCard--TitleHeader"
  >
    Test Score
  </span>
  <div
    className="ScoreCard--CircularProgressbarWrapper"
  >
    <CircularProgressbar
      decorationHue={120}
      maxValue={108}
      text="38.9%"
      value={42}
    />
  </div>
  <a
    href="test.link"
    rel="noreferrer noopener"
    target="_blank"
  >
    How to improve 
    <NewWindowIcon
      isLarge={false}
    />
  </a>
</div>
`;

exports[`ScoreCard renders as expected when score is max 1`] = `
<div
  className="ScoreCard"
>
  <span
    className="ScoreCard--TitleHeader"
  >
    Test Score
  </span>
  <div
    className="ScoreCard--CircularProgressbarWrapper"
  >
    <CircularProgressbar
      decorationHue={120}
      maxValue={108}
      text="100.0%"
      value={108}
    />
  </div>
  <a
    href="test.link"
    rel="noreferrer noopener"
    target="_blank"
  >
    Great! What does this mean 
    <NewWindowIcon
      isLarge={false}
    />
  </a>
</div>
`;

exports[`ScoreCard renders as expected when score is zero 1`] = `
<div
  className="ScoreCard"
>
  <span
    className="ScoreCard--TitleHeader"
  >
    Test Score
  </span>
  <div
    className="ScoreCard--CircularProgressbarWrapper"
  >
    <CircularProgressbar
      backgroundHue={0}
      decorationHue={120}
      maxValue={108}
      text="0.0%"
      value={0}
    />
  </div>
  <a
    href="test.link"
    rel="noreferrer noopener"
    target="_blank"
  >
    How to improve 
    <NewWindowIcon
      isLarge={false}
    />
  </a>
</div>
`;
